src/mock-event-bus/index.ts

```diff
@@ -1,64 +1,41 @@
-import { EventBus, Event } from '../event-bus';
-import { Option, None, Some } from 'funfix';
-
+import { EventBus, Event, EventType } from '../event-bus';
 export type AnyEvent = Event<object>;
 export type AnyHandler = (ev: AnyEvent) => Promise<boolean>;
 
-// TODO: Re-look at all of this file. Not sure where this came from, looks commented differently.
-// Can this be moved into testing
-
 /**
- * Mocks out the EventBus for use in tests.
- *
- * @export
- * @class MockEventBus
- * @implements {EventBus}
+ * MockEventBus is defined here for use during testing across services that use
+ * the event-bus. Examples are: audit and continuum-auth.
  */
 
-// Should be InProcessEventBus
 export class MockEventBus implements EventBus {
-    private queues: Option<Map<string, AnyHandler>> = None;
+    private queues: Map<string, AnyHandler> = new Map();
+    private eventsToHandle: EventType[];
+    private serviceName: string;
 
-    // This lint suppression wouldn't be needed if these params were passed into constructor
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    public async init(_defs: string[], _serviceName: string): Promise<this> {
-        this.queues = Some(new Map());
+    public async register(eventsToHandle: EventType[], serviceName: string): Promise<this> {
```

Comment:
What was the decision behind not using `constructor` again? I think there may be a chance of confusion between `register` and `subscribe` methods on this object

Reply:
That's a good point I'd not thought of ... there was a reason for not using the constructor ... let me try and remember that

Reply:
@Cooryd Done